### PR TITLE
Image record

### PR DESCRIPTION
**Pull request by mariadeck**
Fixes issues #298, #299, #254 

Fix for issue #254 assumes that `imageProcessingErrors` field is set.

### CODE REVIEW

**Comment by ewingrj** [comment]
Deploy preview for *geome-develop* ready!

Built with commit 8efb537f8367d67ab91ba1e27bc9dae09f2648a9

https://deploy-preview-301--geome-develop.netlify.com

**Comment by ewingrj** [comment]
can you paste links to photos so I can easily see the changes please :)

**Comment by mariadeck** [comment]
Invalid Original Url, with bootstrap alert: 
https://deploy-preview-301--geome-develop.netlify.com/record/ark:/99999/fk4Eid2heythere:)

Invalid Original Url, however, `imageProcessingErrors` not set 
https://deploy-preview-301--geome-develop.netlify.com/record/ark:/99999/fk4Eid2invalid

Valid Origin url, photo now displays:
https://deploy-preview-301--geome-develop.netlify.com/record/ark:/99999/fk4Eid2second

File: src/app/views/record/record.html [hunk 1]
```diff
@@ -2,12 +2,37 @@
...
+      ng-if="$ctrl.invalidUrl && $ctrl.record.entity === 'Sample_Photo'"
+      class="alert alert-danger"
+      role="alert"
+    >
+      <i class="fa fa-exclamation-circle"></i>
+      Original Url is invalid
+    </span>
+    <span
+      ng-if="$ctrl.record.processed === 'false' && $ctrl.record.entity === 'Sample_Photo'"
+      class="alert alert-danger"
```
Comment (ewingrj) [inline src/app/views/record/record.html]: maybe alert-warning or alert-info would be better. danger indicates an error

File: src/app/views/record/record.html [hunk 2]
```diff
@@ -2,12 +2,37 @@
...
+      <i class="fa fa-exclamation-circle"></i>
+      Image is still being processed
+    </span>
+  </h3>
 
   <div class="main-details" layout="row" layout-align="center center">
     <div ng-if="$ctrl.record.img1024" flex="100">
       <a href="{{ $ctrl.record.img1024 }}"><img ng-src="{{ $ctrl.record.img512 }}"/></a>
     </div>
+    <div
```
Comment (ewingrj) [inline src/app/views/record/record.html]: this section is not needed. We don't want to support base64 encoded urls for the image.

I made a bug fix in the backed that should prevent this from showing 'processed=true' and specify an error instead.

this entire section can be removed

File: src/app/views/record/record.html [hunk 3]
```diff
@@ -42,7 +67,9 @@ <h3>{{ $ctrl.record.entity }} {{ $ctrl.getIdentifier($ctrl.record) }}</h3>
     <table flex>
       <tbody>
         <tr ng-repeat="(key, value) in $ctrl.auxiliaryRecordDetails(0) track by key">
-          <th>{{ key }}:</th>
+          <th ng-class="{'error': !value.href && $ctrl.record.entity === 'Sample_Photo'}">
```
Comment (ewingrj) [inline src/app/views/record/record.html]: this logic will make any attribute that isn't have a href contain the error class. This is not what we want. Better to just explicitly check `key==='imageProcessingErrors'`

File: src/style/fims/_record.scss [hunk 4]
```diff
@@ -24,6 +24,10 @@
   td.sequence {
     word-break: break-all;
   }
+
+  .error {
```
Comment (ewingrj) [inline src/style/fims/_record.scss]: I think just using `text-danger` class is better. This keeps the error colors consistent throughout the app

File: src/app/views/photo-upload/photo-upload.html [hunk 5]
```diff
@@ -31,7 +31,7 @@ <h2 class="page-header">Upload Photos</h2>
         <strong>OPTION 2: Upload photos from your hard-drive.</strong>
       </p>
 
-      <p>The GEOME bulk image loader accepts a directory of images zipped into a single file.  The max file size is 2GB. If you need to upload more then that, split the upload into multiple uploads.  When uploading images, you have the following options:</p>
+      <p>The GeOMe bulk image loader accepts a directory of images zipped into a single file.  The max file size is 2GB. If you need to upload more than that, split the upload into multiple uploads.  When uploading images, you have the following options:</p>
```
Comment (ewingrj) [inline src/app/views/photo-upload/photo-upload.html]: GEOME please ;)

File: src/app/views/record/record.html (hunk 3 above)
Comment (ewingrj) [reply src/app/views/record/record.html]: this also needs to work w/ event photos

File: src/app/views/record/record.html (hunk 2 above)
Comment (mariadeck) [reply src/app/views/record/record.html]: Ok, I'm confused on this one. I'm testing on photos that are already loaded, and removing any of  this section makes images not appear, and still shows 'processed=true'

Comment (ewingrj) [reply src/app/views/record/record.html]: that's b/c those photos we loaded before I fixed a bug in the photo processing. Try and load those again and it should have an error

**Comment by mariadeck** [comment]
https://deploy-preview-301--geome-develop.netlify.com/record/ark:/99999/fk4Eid2base64
https://deploy-preview-301--geome-develop.netlify.com/record/ark:/99999/fk4Eid2validUrl

File: src/app/views/record/record.html (hunk 2 above)
Comment (mariadeck) [reply src/app/views/record/record.html]: Ok, got rid of the duplicate. all of my testers were base64 which is why i couldnt see the img512 before

**Comment by mariadeck** [comment]
https://deploy-preview-301--geome-develop.netlify.com/record/ark:/99999/fk4Eid2localUpload

**Pull request merged**